lms/static/js/bookmarks/views/bookmarks_list.js

```diff
@@ -0,0 +1,124 @@
+                _.bindAll(this, 'render', 'breadcrumbTrail', 'userFriendlyDate', 'bookmarkUrl');
+            },
+
+            render: function () {
+                var data = {
+                    bookmarks: this.collection.models,
+                    breadcrumbTrail: this.breadcrumbTrail,
+                    userFriendlyDate: this.userFriendlyDate,
+                    bookmarkUrl: this.bookmarkUrl
+                };
+                this.$el.html(this.template(data));
+                this.delegateEvents();
+                return this;
+            },
+
+            loadBookmarks: function () {
+                var view = this;
+
+                this.hideErrorMessage();
+                this.showBookmarksContainer();
+                this.showLoadingMessage();
+
+                this.collection.url = this.url;
+                this.collection.fetch({
+                    reset: true,
+                    data: {course_id: this.courseId, fields: 'path'}
+                }).done(function () {
+                    view.hideLoadingMessage();
+                    view.render();
+                    view.focusBookmarksElement();
+                }).fail(function () {
+                    view.hideLoadingMessage();
+                    view.showErrorMessage();
+                });
+            },
+
+            visitBookmark: function (event) {
+                window.location = event.target.pathname;
+            },
+
+            breadcrumbTrail: function (bookmarkPath, unitDisplayName) {
+                var separator = ' <i class="icon fa fa-caret-right" aria-hidden="true"></i><span class="sr">-</span> ';
+                var names = _.pluck(bookmarkPath, 'display_name');
+                names.push(unitDisplayName);
+                return names.join(separator);
+            },
+
+            userFriendlyDate: function (isoDate) {
+                moment.locale(this.langCode);
+                return moment(isoDate).format('LL');
+            },
+
+            bookmarkUrl: function (courseId, usageId) {
+                return '/courses/' + courseId + '/jump_to/' + usageId
+            },
+
+            isVisible: function () {
+                return this.$('#my-bookmarks').is(":visible");
+            },
+
+            hideBookmarks: function () {
+              this.$el.hide();
+              $(this.coursewareContentElement).show();
+            },
+
+            showBookmarksContainer: function () {
+                $(this.coursewareContentElement).hide();
+                // Empty el if there anything in it so that we are in clean state.
+                this.$el.html('');
+                this.$el.show();
+            },
+
+            showLoadingMessage: function () {
+                this.loadingMessageView.showMessage(this.loadingMessage, this.loadingIcon);
+            },
+
+            hideLoadingMessage: function () {
+                this.loadingMessageView.hideMessage();
+            },
+
```

Comment:
can we merge `showLoadingMessage` and  `hideLoadingMessage` into a  single method by passing a boolean flag
and then based on the boolean; we can call the specific method.